src/comet_llm/chains/state.py

```diff
@@ -58,6 +59,7 @@ def global_chain_exists() -> bool:
     return _APP_STATE.chain_exists()
 
 
+@exceptions.filter(allow_raising=config.raising_enabled(), summary=app.SUMMARY)
```

Comment:
I was thinking about this decorator.
If we start abusing it everywhere we want it will become a problem soon to understand the exception-related logic in the SDK, especially if it's config dependent.
Let's use this decorator (and in general the logic for deciding between logging and raising) only for the public API level. I think we will benefit from it in the future.

So what I propose here is this:
```python
# inside State class
    @property
    def chain(self) -> Optional["chain.Chain"]:
        result: "chain.Chain" = self._thread_context_registry.get("global-chain")
        return result

def get_global_chain() -> Optional["chain.Chain"]:
    result = _APP_STATE.chain
```

Let's raise an exception directly from methods like end_chain (which can now be decorated with filter decorator) and from Span class.